### PR TITLE
Change auto-dithering to DitherType.ORDERED and ERROR_DIFFUSION where applicable

### PR DESCRIPTION
**Pull request by LightArrowsEXE**
Courtesy of @wiwaz. From our PMs:

@wiwaz:
```
i did tests
at high bitdepths the difference between ordered and error diffusion spatially is unnoticeable
but ordered is static
using error diffusion all the time is hurting our compressability
and maybe even making quality worse
since its adding random temporal noise that we really dont even need
again, the difference is unnoticeable spatially at high depths
ordered only looks ugly when dithered down to 8
so basically when we are doing 32 to 16 or 16 to 10 whatever we should really be considering ordered instead
and opting for error diffusion only for the final dither down
```

With this in mind, this PR is more of a call to action for additional testers to see how well this works out in real-life application. Please test! @NSQY @Setsugennoao

### CODE REVIEW

**Comment by Setsugennoao** [comment]
Should be based on input and output bitdepth, not just blindly use ordered (which i don't like tbh...)

**Comment by LightArrowsEXE** [comment]
Courtesy of @NSQY (PLEASE POST HERE >:((( )

```shell
<html>
<body>
<!--StartFragment-->

1 | >>> from vstools import core, vs, clip_async_render
-- | --
2 | >>> core.num_threads = 1
3 | >>> clip = core.std.BlankClip(None, 3840, 2160, format=vs.YUV444PS, length=500, keep=True
4 | ... )
5 | >>> clip_async_render(clip.resize.Point(format=vs.YUV444P16, dither_type='none'), None, 'Test')
6 | Test ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 500/500 100.00% 257.48 fps 0:00:00
7 | >>> clip_async_render(clip.resize.Point(format=vs.YUV444P16, dither_type='ordered'), None, 'Test')
8 | Test ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 500/500 100.00% 259.63 fps 0:00:00
9 | >>> clip_async_render(clip.resize.Point(format=vs.YUV444P16, dither_type='error_diffusion'), None, 'Test')
10 | Test ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 500/500 100.00% 53.58 fps 0:00:00

<!--EndFragment-->
</body>
</html>

**Comment by wiwaz** [comment]
> Should be based on input and output bitdepth, not just blindly use ordered (which i don't like tbh...)

Ordered specifically is terrible for dithering down to 8 bits so yes it'd have to be dependant on in/out bits unless a more robust dithering algorithm (void?) is used.

**Comment by LightArrowsEXE** [comment]
Sorted in #85.
